docs/versioned_docs/version-zenith/developer-guide/typescript/snippets/custom-types/index.ts

```diff
@@ -0,0 +1,48 @@
+import { dag, object, func, GitRepository, field } from '@dagger.io/dagger';
+
+@object()
+class GitHubAccount {
+  @field()
+  username: string;
+
+  @field()
+  email: string;
+
+  @field()
+  url: string
+
+  constructor(username: string, email: string) {
+    this.username = username;
+    this.email = email;
+    this.url = `https://github.com/${username}`
+  }
+}
+
+@object()
+class GitHubOrganization {
+  @field()
+  url: string;
+
+  @field()
+  repository: GitRepository[];
+
+  @field()
+  members: GitHubAccount[];
+}
+
+@object()
+class MyModule {
```

Comment:
Since this seems to be a module reated to github I think the module name should be "github", making the main object "Github". Notice I didn't suggest "GitHub", since that would make the module name `git-hub` and constructor function `dag.gitHub()`.

Note that "custom types" are prefixed with the module's name in the API to avoid conflicts, so **we don't want users to add the prefix themselves**. In this case, current types will be:
- MyModule
- MyModuleGitHubAccount
- MyModuleGitHubOrganization

In the Python version for this doc I renamed `MyModule` to `Github` and removed the prefixes (i.e., just `Account` and `Organization`) for this reason.

Renaming without removing the prefixes, will get you these types:
- Github
- GithubGitHubOrganization
- GithubGitHubAccount

Reply:
Thanks for this, makes perfect sense. I have made the change for Go and TypeScript in https://github.com/dagger/dagger/pull/6681/commits/d08bafd3ffae799421a7fa8e71cf6ead84eca99e but would like to have your eyes on it @helderco @jedevc @TomChv @vito - thanks!